### PR TITLE
Update slices-pattern.md

### PR DESCRIPTION
**Pull request by dbritto-dev**
## Related Issues or Discussions

Improving docs

## Summary

The current guide could misguide people in sense that they could use create slice functions to access functions but also any non function which is wrong

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Comment by norama** [comment]
Super, just please include typescript example as well.

**Comment by dbritto-dev** [comment]
> Super, just please include typescript example as well.

we already had a ts guide -> https://docs.pmnd.rs/zustand/guides/typescript

**Comment by norama** [comment]
> > Super, just please include typescript example as well.
> 
> we've already a ts guide -> https://docs.pmnd.rs/zustand/guides/typescript

Yes, I have seen that as well, just please include this `get()` access as well as some explanation for 
```javascript
StateCreator<
  BearSlice & FishSlice,
  [],
  [],
  FishSlice
>
```
here:
https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern

**Comment by dbritto-dev** [comment]
> https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern

sure

**Review by dai-shi — COMMENTED** [review]
Looks okay to me, but leaving others to review.

File: docs/guides/typescript.md [hunk 1]
```diff
@@ -392,9 +399,26 @@ const createFishSlice: StateCreator<
...
 
-const useBoundStore = create<BearSlice & FishSlice>()((...a) => ({
+const createSharedSlice: StateCreator<
+  BearSlice & FishSlice,
+  [],
+  [],
+  SharedSlice
+> = (set, get) => ({
+  addBoth: () => {
+    // reusing previous methods
```
Comment (sewera) [inline docs/guides/typescript.md:409]:
```suggestion
    // you can reuse previous methods
```

File: docs/guides/typescript.md [hunk 2]
```diff
@@ -392,9 +399,26 @@ const createFishSlice: StateCreator<
...
+    get().addBear()
+    get().addFish()
+    // or doing from scratch
```
Comment (sewera) [inline docs/guides/typescript.md:412]:
```suggestion
    // or do them from scratch
```

**Review by sewera — APPROVED** [review]
Looks good :)

**Pull request merged**